browser/brave_content_browser_client.cc

```diff
@@ -223,6 +230,35 @@ void BraveContentBrowserClient::AdjustUtilityServiceProcessCommandLine(
 #endif
 }
 
+std::vector<std::unique_ptr<blink::URLLoaderThrottle>>
+BraveContentBrowserClient::CreateURLLoaderThrottles(
+    const network::ResourceRequest& request,
+    content::BrowserContext* browser_context,
+    const base::RepeatingCallback<content::WebContents*()>& wc_getter,
+    content::NavigationUIData* navigation_ui_data,
+    int frame_tree_node_id) {
+  auto result = ChromeContentBrowserClient::CreateURLLoaderThrottles(
+        request, browser_context, wc_getter, navigation_ui_data,
+        frame_tree_node_id);
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+  const auto* cmd_line = base::CommandLine::ForCurrentProcess();
+  if (cmd_line->HasSwitch(speedreader::kEnableSpeedreader)) {
+    // Work only with casual main frame navigations.
+    if (request.url.SchemeIsHTTPOrHTTPS() &&
+        request.resource_type ==
+        static_cast<int>(content::ResourceType::kMainFrame)) {
+      // Note that we check the whitelist before any redirects, while distilling
+      // will be performed on a final document (the last in the redirect chain).
+      if (speedreader::IsWhitelisted(request.url)) {
+        result.push_back(std::make_unique<speedreader::SpeedReaderThrottle>(
```

Comment:
wasn't sure I followed the code correctly: SpeedReaderThrottle only applies to the `kMainFrame` URLs, any other requests discovered early get passed through? or throttled as well?

Reply:
hmm, what other requests exactly?

Reply:
was thinking of preload/preconnect/etc, but on second thought these wouldn't get discovered  until the main frame document proceeds down the pipeline